From ce3b1403bd88261b5461a9dcb7d6d6be9185703e Mon Sep 17 00:00:00 2001
From: Richard Purdie <richard.purdie@linuxfoundation.org>
Date: Wed, 1 May 2019 22:15:03 +0100
Subject: [PATCH] Fix error on musl:

| ../../elfutils-0.176/tests/elfstrmerge.c: In function 'main':
| ../../elfutils-0.176/tests/elfstrmerge.c:370:60: error: 'ALLPERMS' undeclared (first use in this function); did you mean 'EPERM'?
|        fdnew = open (fnew, O_WRONLY | O_CREAT, st.st_mode & ALLPERMS);
|                                                             ^~~~~~~~
|                                                             EPERM
| ../../elfutils-0.176/tests/elfstrmerge.c:370:60: note: each undeclared identifier is reported only once for each function it appears in

Upstream-Status: Inappropriate [workaround in musl]

Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>

---
 tests/elfstrmerge.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tests/elfstrmerge.c b/tests/elfstrmerge.c
index abbdf3f..bd90f4d 100644
--- a/tests/elfstrmerge.c
+++ b/tests/elfstrmerge.c
@@ -33,6 +33,11 @@
 #include ELFUTILS_HEADER(dwelf)
 #include "elf-knowledge.h"
 
+/* for musl */
+#ifndef ALLPERMS
+# define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)/* 07777 */
+#endif
+
 /* The original ELF file.  */
 static int fd = -1;
 static Elf *elf = NULL;
